lib/private/Files/View.php

```diff
@@ -1183,7 +1183,9 @@ private function basicOperation($operation, $path, $hooks = [], $extraParam = nu
 					$this->removeUpdate($storage, $internalPath);
 				}
 				if ($result && in_array('write', $hooks,  true) && $operation !== 'fopen' && $operation !== 'touch') {
-					$this->writeUpdate($storage, $internalPath);
+					$size = $operation === 'mkdir' ? 0 : null;
+					$size = $operation === 'file_put_contents' ? $result : $size;
```

Comment:
```suggestion
					$size = null;
					if ($operation === 'mkdir') {
						$size = 0;
					} elseif ($operation === 'file_put_contents') {
						$size = $result;
					}
```
Perhaps a clearer control flow :)